tests/internal/test_settings.py

```diff
@@ -248,3 +257,35 @@ def test_remoteconfig_logs_injection_jsonlogger(run_python_code_in_subprocess):
     log_enabled, log_disabled = map(json.loads, err.decode("utf-8").strip().split("\n")[0:2])
     assert log_enabled["dd.trace_id"] == trace_id
     assert "dd.trace_id" not in log_disabled
+
+
+def test_remoteconfig_header_tags(run_python_code_in_subprocess):
+    env = os.environ.copy()
+    env.update({"DD_TRACE_HEADER_TAGS": "X-Header-Tag-419:env_set_tag_name"})
+    out, err, status, _ = run_python_code_in_subprocess(
+        """
+from ddtrace import config, tracer
+from ddtrace.contrib import trace_utils
+from tests.internal.test_settings import _base_rc_config
+
+with tracer.trace("test") as span:
+    trace_utils.set_http_meta(span,
+                              config.falcon,  # randomly chosen http integration config
+                              request_headers={"X-Header-Tag-420": "foobarbanana", "X-Header-Tag-419": "helloworld"})
+assert span.get_tag("header_tag_420") is None
+assert span.get_tag("env_set_tag_name") == "helloworld"
+
+config.http._reset()
+config._header_tag_name.invalidate()
+config._handle_remoteconfig(_base_rc_config({"tracing_header_tags": ["X-Header-Tag-420:header_tag_420"]}))
+
+with tracer.trace("test_rc_override") as span2:
+    trace_utils.set_http_meta(span2,
+                              config.falcon,  # randomly chosen http integration config
+                              request_headers={"X-Header-Tag-420": "foobarbanana", "X-Header-Tag-419": "helloworld"})
+assert span2.get_tag("header_tag_420") == "foobarbanana", span2._meta
+assert span2.get_tag("env_set_tag_name") is None
```

Comment:
I think it's worth unsetting the configuration

```suggestion
assert span2.get_tag("env_set_tag_name") is None
config._handle_remoteconfig(_base_rc_config({}))
```

and ensuring the previous behaviour is maintained

Reply:
Let me know if the current state of this test matches what you're looking for.

Reply:
yep, it is, thanks!